gef.py

```diff
@@ -7386,6 +7550,7 @@ def context_regs(self):
         return
 
     def context_stack(self):
+        global DEREFERENCING_STACK
```

Comment:
What is this for? Looks like it's some command context state. This can hopefully be cleaned up, and shouldn't be a global, especially a capitalized one, which implies it's a constant.

Reply:
@hugsy I don't know how to resolve this issue. The `DEREFERENCING_STACK` variable needs to be cleaned up indeed. For, as you can see, it's a dirty hack to record that the dereference was invoked by the `ContextCommand::context_stack()` function.
```diff
@@ -7397,7 +7564,9 @@ def context_stack(self):
                mem = read_memory(sp, 0x10 * nb_lines)
                gef_print(hexdump(mem, base=sp))
            else:
+               DEREFERENCING_STACK = True
                gdb.execute("dereference {:#x} l{:d}".format(sp, nb_lines))
+               DEREFERENCING_STACK = False
```

The reason I need this information is that `process_lookup_address()` cannot correctly recognize the addresses belonging to the stack, so an additional condition is needed to avoid error. Why can't `process_lookup_address()` recognize the addresses properly? Because it utilizes `/proc/pid/maps` to determine where an address resides in, which won't work on Cygwin or Windows platform.
```diff
@@ -8167,7 +8336,7 @@ def do_invoke(self, argv):
            return

        addr = int(addr)
-       if process_lookup_address(addr) is None:
+       if process_lookup_address(addr) is None and not DEREFERENCING_STACK:
            err("Unmapped address")
            return
```

One possible solution, in my opinion, is using some heuristics to check the type of pointers. Interestingly, **peda-gdb** can dereference pointers correctly (in most cases), so maybe its strategy can provide some insights.